File: WorkQueues/python/receiver/src/worker.py
import pika
import sys
import time

QUEUE_NAME = 'task_queue'

connection = pika.BlockingConnection(
    pika.ConnectionParameters(host='rabbitmq')
)

channel = connection.channel()

channel.queue_declare(queue=QUEUE_NAME, durable=True)

print('[PY] Waiting for messages. To exit press Ctrl + C')

def callback(ch, method, properties, body):
    print("[PY] Received %r" % body.decode())
    time.sleep(body.count(b'.'))
    print('[PY] Done!')
    ch.basic_ack(delivery_tag=method.delivery_tag)

channel.basic_qos(prefetch_count=1)
channel.basic_consume(queue=QUEUE_NAME, on_message_callback=callback)

channel.start_consuming()

File: WorkQueues/php/receiver/src/worker.php
<?php

require_once __DIR__ . '/vendor/autoload.php';
use PhpAmqpLib\Connection\AMQPStreamConnection;

const QUEUE_NAME = 'task_queue';

$connection = new AMQPStreamConnection('rabbitmq', 5672, 'guest', 'guest');
$channel = $connection->channel();

$channel->queue_declare(QUEUE_NAME, false, true, false, false);

echo " [PHP] Waiting for messages. To exit send an \"exit\" message.\n";

$callback = function($msg) {
    echo '[PHP] Received ', $msg->body, "\n";
    sleep(substr_count($msg->body, '.'));
    echo "[PHP] Done!\n";
    $msg->ack();

    if($msg->body == 'exit') {
        echo "[PHP] Received an 'exit' message. Leaving...\n";
        $msg->getChannel()->close();
    }
};

$channel->basic_qos(null, 1, null);
$channel->basic_consume(QUEUE_NAME, '', false, false, false, false, $callback);

while($channel->is_open()) {
    $channel->wait();
}

echo "The channel was closed.\n";

$connection->close();

echo "Connection closed. Leaving worker.\n";